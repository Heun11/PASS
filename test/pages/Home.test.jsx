import React from 'react';
import { render, cleanup, screen } from '@testing-library/react';
import { expect, it, afterEach, describe } from 'vitest';
import { Home } from '@pages';
import createMatchMedia from '../helpers/createMatchMedia';

describe('Home Page', () => {
  afterEach(cleanup);
  // if tests are failing check this value vs value in Home.jsx
  const h1FontSize = '144px';

  it('renders desktop', () => {
    render(<Home />);
    const h1 = screen.getByTestId('testHomeH1');
    const h1Styles = getComputedStyle(h1);

    expect(screen.getByTestId('testHomeSection')).toBeDefined();
    expect(h1Styles.fontSize).toBe(h1FontSize);
  });

  it('renders mobile', () => {
    window.matchMedia = createMatchMedia(599);
    render(<Home />);
    const h1 = screen.getByTestId('testHomeH1');
    const h1Styles = getComputedStyle(h1);

    expect(screen.getByTestId('testHomeSection')).toBeDefined();
    expect(h1Styles.fontSize).not.toBe(h1FontSize);
  });
});
