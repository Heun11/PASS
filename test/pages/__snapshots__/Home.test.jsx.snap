// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home Page > renders 1`] = `
<div>
  <main
    class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      class="MuiBox-root css-1gqbj6n"
    >
      <section
        id="home"
      >
        <div
          class="MuiStack-root css-le9p6c-MuiStack-root"
        />
      </section>
    </div>
  </main>
</div>
`;
